language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/DefaultValueGenerationUtil.java

```diff
@@ -292,22 +305,51 @@ private static Optional<String> getDefaultValueForType(TypeSymbol bType, boolean
                 break;
             case SINGLETON:
                 valueString = rawType.signature();
-                valueString = isSnippet ? generateSnippetEntry(valueString, offset) : valueString;
+                valueString =
+                        isSnippet ? generateSnippetEntry(valueString, context.getPlaceholderCount()) : valueString;
+                context.incrementPlaceholderCount();
                 break;
             case JSON:
             case NIL:
             case ANY:
                 valueString = "()";
-                valueString = isSnippet ? generateSnippetEntry(valueString, offset) : valueString;
+                valueString =
+                        isSnippet ? generateSnippetEntry(valueString, context.getPlaceholderCount()) : valueString;
+                context.incrementPlaceholderCount();
                 break;
             default:
                 Optional<String> value = getDefaultValueForTypeDescKind(typeKind);
                 if (value.isEmpty()) {
                     return value;
                 }
-                return isSnippet ? Optional.of(generateSnippetEntry(value.get(), offset)) : value;
+                valueString =
+                        isSnippet ? generateSnippetEntry(value.get(), context.getPlaceholderCount()) : value.get();
+                context.incrementPlaceholderCount();
         }
 
         return Optional.of(valueString);
     }
+
+    /**
+     * Represents a context to track the number of placeholders.
+     */
+    public static class SnippetContext {
+        private int placeholderCount;
+
+        public SnippetContext() {
+            placeholderCount = 1;
+        }
+
+        public SnippetContext(int placeholderCount) {
```

Comment:
There was a scenario where the count was set to -1 instead of 0, thats why this constructor was added. Changed -1 to 0 and ran the test cases and none of the test cases failed. So will remove this.